apollo/command_to_gcode.py

```diff
@@ -0,0 +1,145 @@
+import asyncio
+
+from apollo.smoothie import SmoothieCom
+from config.settings import Config
+
+
+class CommandToGCode(object): 
+    """
+    this module converts front-end commands to GCode
+    then sends GCode strings to Smoothie-Com module
+    """
+
+    def __init__(self):
+        self.smoothie_com = SmoothieCom(Config.SMOOTHIE_URL, Config.SMOOTHIE_PORT)
+
+    @asyncio.coroutine
+    def connect(self):
+        """
+        tell the smoothie module to connect to the motor controller
+        return True if successful
+        """
+        return (yield from self.smoothie_com.connect())
+
+    @asyncio.coroutine
+    def send_gcode(self,string):
+        """
+        send a GCode string to the smoothie module
+        """
+        if len(string):
+            return (yield from self.smoothie_com.send(string))
+
+    @asyncio.coroutine
+    def process(self, command):
+        """
+        receive front-end commands, and convert them to GCode
+        returns None or a coordinate dict
+        """
+
+        # check to see if this module has a method matching the command's 'type'
+        method = getattr(self, command.get('type', '') , None)
```

Comment:
I was thinking, instead of directly checking whether or not the instance variable has a method with the name of the type attribute, you can register all such methods in a dictionary.. E.g....

```
COMMAND_TYPE_TO_HANDLER = {
'move': self.move,
....
}

# then do

try:
    command_type:
    handler = COMMAND_TYPE_TO_HANDLER[command_type]
    return handler(command)
   except KeyError:
         log.error('invalid command....)
```